File: backend.scala/test/com/skewwhiffy/auraltester/clefs/ClefTest.scala
package com.skewwhiffy.auraltester.clefs

import com.skewwhiffy.auraltester.testutils.TestData
import org.scalatest.flatspec.AnyFlatSpec

class ClefTest extends AnyFlatSpec {
  private val abc = TestData.random.string
  private val lowLedgerNote = TestData.random.absoluteNote
  private val highLedgerNote = TestData.random.absoluteNote

  it should "retain abc, lowLedgerNote and highLedgerNote" in {
    val actual = Clef(abc, lowLedgerNote, highLedgerNote)

    assert(actual.abc == abc)
    assert(actual.lowLedgerNote == lowLedgerNote)
    assert(actual.highLedgerNote == highLedgerNote)
  }

  it should "equate same clefs" in {
    val first = Clef(abc, lowLedgerNote, highLedgerNote)
    val second = Clef(abc, lowLedgerNote, highLedgerNote)

    assert(first == second)
  }
}

File: backend.kotlin/src/test/kotlin/com/skewwhiffy/notation/model/clef/ClefTest.kt
package com.skewwhiffy.notation.model.clef

import com.skewwhiffy.notation.model.clef.Clef
import com.skewwhiffy.test.util.TestData
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ClefTest {
  private val abc = TestData.random.string
  private val lowLedgerNote = TestData.random.absoluteNote
  private val highLedgerNote = TestData.random.absoluteNote

  @Test
  fun `retain abc, lowLedgerNote and highLedgerNote`() {
    val actual = Clef(abc, lowLedgerNote, highLedgerNote)

    assertThat(actual.abc).isEqualTo(abc)
    assertThat(actual.lowLedgerNote).isEqualTo(lowLedgerNote)
    assertThat(actual.highLedgerNote).isEqualTo(highLedgerNote)
  }

  @Test
  fun `equate same clefs`() {
    val first = Clef(abc, lowLedgerNote, highLedgerNote)
    val second = Clef(abc, lowLedgerNote, highLedgerNote)

    assertThat(first).isEqualTo(second)
  }
}